modules/users/server/config/strategies/facebook.js

```diff
@@ -36,6 +37,18 @@ module.exports = function (config) {
 
       // Save the user OAuth profile
       users.saveOAuthUserProfile(req, providerUserProfile, done);
+
+      function generateUsername(profile) {
+        var username = '';
+
+        if (profile.emails) {
+          username = profile.emails[0].value.split('@')[0];
+        } else if (profile.name) {
+          username = profile.name.givenName[0] + profile.name.familyName;
+        }
+
+        return username.toLowerCase() || undefined;
```

Comment:
Also, take a look at https://github.com/meanjs/mean/blob/3064b2882048beebf5ff6fc9c3f9599a5650c764/modules/users/server/controllers/users/users.authentication.server.controller.js#L155 looks like it's already trying to figure out a username for the user.